### ISSUE TITLE
`error-message`: support custom error packages

### ISSUE THREAD

**Issue by christopher-buss** [sindresorhus/eslint-plugin-unicorn#2628]
### Description

The [only-throw-error](https://typescript-eslint.io/rules/only-throw-error/) rule from TypeScript ESLint allows for custom objects to be provided. Due to the nature of the TypeScript environment I am in, I have to resort to using a third-party polyfill Error, rather than the built-in. Due to this, the error-message rule does not work as it doesn't detect my Error as an inbuilt error.

It would be nice to be able to provide custom classes to match the rule against.
E.g.
`{ allow: [{ from: "package", name: "Error", package: "@rbxts/luau-polyfill" }] },`

### Examples

```js
// ❌
import { Error } from "@rbxts/luau-polyfill";

throw new Error();

// ✅
import { Error } from "@rbxts/luau-polyfill";

throw new Error("some message");
```

**Comment by fisker** [comment]
We can't know how custom error is constructed, even with type information.

I think we'll skip this.
